api/rosetta/data_integration_test.go

```diff
@@ -97,6 +111,68 @@ func setupAPI(t *testing.T, db *badger.DB) *rosetta.Data {
 	return controller
 }
 
+func setupRecorder(endpoint string, input interface{}, options ...func(*http.Request)) (*httptest.ResponseRecorder, echo.Context, error) {
+
+	var payload []byte
+
+	switch input.(type) {
+
+	case []byte:
+		payload = input.([]byte)
```

Comment:
... you can avoid this by doing:

```go
switch in := input.(type) {
  case []byte:
    payload = in
}
```

But overall, there is a much simpler way and this switch is uber-complicated.